### PR TITLE
Fix concurrent dagrequest issue

### LINKED ISSUE #713: ci failures with java.util.ConcurrentModificationException
Caused by: java.lang.RuntimeException: java.util.ConcurrentModificationException
	at com.pingcap.tikv.meta.TiDAGRequest.copy(TiDAGRequest.java:1033)
	at org.apache.spark.sql.execution.RegionTaskExec$$anonfun$internalRowToUnsafeRowWithIndex$3$$anonfun$apply$3$$anon$1.call(CoprocessorRDD.scala:264)
	at org.apache.spark.sql.execution.RegionTaskExec$$anonfun$internalRowToUnsafeRowWithIndex$3$$anonfun$apply$3$$anon$1.call(CoprocessorRDD.scala:262)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	... 3 more
Caused by: java.util.ConcurrentModificationException
	at java.util.ArrayList.writeObject(ArrayList.java:770)
	at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:1140)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1496)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178)
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1548)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1509)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:348)
	at com.pingcap.tikv.meta.TiDAGRequest.copy(TiDAGRequest.java:1028)
	... 8 more

### PR DESCRIPTION
**Pull request by birdstorm**
Fix #713

### CODE REVIEW

**Comment by birdstorm** [comment]
/run-all-tests tidb=release-2.1 tikv=release-2.1 pd=release-2.1

File: core/src/main/scala/org/apache/spark/sql/execution/CoprocessorRDD.scala [hunk 1]
```diff
@@ -180,6 +180,7 @@ case class RegionTaskExec(child: SparkPlan,
   private val downgradeThreshold =
     sqlConf.getConfString(TiConfigConst.REGION_INDEX_SCAN_DOWNGRADE_THRESHOLD, "10000").toInt
   private lazy val project = UnsafeProjection.create(schema)
+  private val logger = Logger.getLogger(getClass.getName)
```
Comment (zhexuany) [inline core/src/main/scala/org/apache/spark/sql/execution/CoprocessorRDD.scala]: this may cause cannot be serialized issue

**Pull request merged**